src/coreclr/vm/codeman.cpp

```diff
@@ -4496,7 +4513,6 @@ BOOL ExecutionManager::IsManagedCode(PCODE currentPC, HostCallPreference hostCal
     }
 
     ReaderLockHolder rlh(hostCallPreference);
-    if (!rlh.Acquired())
```

Comment:
By removing the if statement, we are unconditionally going down the failure path, why are you doing this?

Reply:
No, that isn't what I'm saying. In this case, line 4515 unconditionally takes the lock, and then as the if statement is removed, the new code path will unconditionally return FALSE, and set *pfFailedReaderLock to TRUE.

Reply:
I certainly had to remove all the code under the `if` body, not only `if` statement.